src/tests/nurse/test_views.py

```diff
@@ -127,6 +145,92 @@ def s3_mock():
 }
 
 
+@pytest.fixture
+def prescription(user):
+    patient = Patient.objects.create(**patient_data)
+    nurse, _ = Nurse.objects.get_or_create(user=user)
+    nurse.patients.add(patient)
+    return Prescription.objects.create(patient=patient, **prescription_data)
+
+
+@pytest.mark.django_db
+class TestSendEmailToDoctorView:
+    def test_send_email_to_doctor(self, client, prescription):
+        url = reverse_lazy("send-email-to-doctor", kwargs={"pk": prescription.id})
+
+        response = client.post(url)
+        assert response.status_code == status.HTTP_200_OK
+        assert response.data == {"message": "Email sent"}
+        assert len(mail.outbox) == 1
+        assert mail.outbox[0].to == [prescription.email_doctor]
+
+    def test_send_email_to_doctor_404(self, client):
+        url = reverse_lazy("send-email-to-doctor", kwargs={"pk": 1})
+        response = client.post(url)
+        assert response.status_code == status.HTTP_404_NOT_FOUND
+
+    def test_doctor_has_no_email(self, client, prescription):
+        prescription.email_doctor = ""
+        prescription.save()
+        url = reverse_lazy("send-email-to-doctor", kwargs={"pk": prescription.id})
+        response = client.post(url)
+        assert response.status_code == status.HTTP_400_BAD_REQUEST
+        assert response.data == {"error": "Doctor has no email"}
+
+    def test_user_without_name(self, client, prescription, user):
+        user.first_name = ""
+        user.last_name = ""
+        user.save()
+        url = reverse_lazy("send-email-to-doctor", kwargs={"pk": prescription.id})
+        response = client.post(url)
+        assert response.status_code == status.HTTP_400_BAD_REQUEST
+        assert response.data == {
+            "error": "Please update your profile with your first and last name"
+        }
+
+    def test_send_email_to_doctor_500(self, client, prescription):
+        url = reverse_lazy("send-email-to-doctor", kwargs={"pk": prescription.id})
+
+        with patch("nurse.views.send_mail", side_effect=Exception("SMTP Error")):
+            response = client.post(url)
+
+            assert response.status_code == status.HTTP_500_INTERNAL_SERVER_ERROR
+            assert response.data == {"error": "SMTP Error"}
+
+    def test_send_email_to_doctor_user_not_nurse(
+        self, client, prescription, another_user
+    ):
+        url = reverse_lazy("send-email-to-doctor", kwargs={"pk": prescription.id})
+
+        # Authenticate as another user who is not associated with the patient
+        client.force_authenticate(user=another_user)
+
+        response = client.post(url)
+        assert response.status_code == status.HTTP_403_FORBIDDEN
+        assert response.data == {
+            "error": "You are not authorized to send emails for this patient"
+        }
```

Comment:
maybe prove that there're no nurse associated then if we want to cover this case.
The other approach would have to get_or_create the nurse like I suggested in the view so there's always a nurse

Reply:
I'm testing to see if a patient is on this nurse's list.
- if another nurse wants to send an e-mail for this patient, then there's an error: only the nurse who has the patient in her list should be able to send the renewal